tests/channel_test.c

```diff
@@ -710,3 +714,153 @@ static int s_test_channel_cancels_pending_tasks(struct aws_allocator *allocator,
 }
 
 AWS_TEST_CASE(channel_cancels_pending_tasks, s_test_channel_cancels_pending_tasks)
+
+struct channel_connect_test_args {
+    struct aws_mutex *mutex;
+    struct aws_condition_variable cv;
+    int error_code;
+    struct aws_channel *channel;
+    bool setup;
+    bool shutdown;
+};
+
+static void s_test_channel_connect_some_hosts_timeout_setup(
+    struct aws_client_bootstrap *bootstrap,
+    int error_code,
+    struct aws_channel *channel,
+    void *user_data) {
+
+    struct channel_connect_test_args *test_args = user_data;
+    test_args->setup = true;
```

Comment:
this probably needs to acquire the mutex before mutating.

Reply:
Fixed.

Reply:
fixed